src/main/java/com/squareup/javapoet/MethodSpec.java

```diff
@@ -111,13 +106,22 @@ void emit(CodeWriter codeWriter, String enclosingName, Set<Modifier> implicitMod
         firstException = false;
       }
     }
-    codeWriter.emit(" {\n");
 
-    codeWriter.indent();
-    codeWriter.emit(code);
-    codeWriter.unindent();
+    if (hasModifier(Modifier.ABSTRACT)) {
+      codeWriter.emit(";\n");
+    } else if (hasModifier(Modifier.NATIVE)) {
+      // code is allowed to support stuff like GWT JSNI
```

Comment:
super nit: Capitalize + period. [Comments should be sentences][1].

 [1]: nedbatchelder.com/blog/201401/comments_should_be_sentences.html